wonderunit/storyboarder
# Each board can have a "link" to a source file from an editor (e.g.: PSD)

Storyboarder should know what source file is linked per board (e.g.: file path to PSD).

- [x] "Open In Editor" will add the PSD path of the created PSD to the JSON file.
- [x] If an artist begins to draw on a linked board, warn first about link being broken.
- - [x] If artist agrees to continue after the warning, remove linkage in JSON. PSD file remains on filesystem until cleanup task is run.
- [x] "Open in Editor" should warn before overwriting an existing PSD.
- [x] "Clean Up Scene" task should not delete PSD files if they are _actively_ linked.
- [x] Clear filesystem watcher on window close

To test:
- [x] Test that thumbnails update (for both currently selected board and not)
- [x] Test closing and re-opening project
- [x] Test closing and re-opening Storyboarder
- [x] Monitor the JSON data and make sure it's changing properly
- [x] Monitor the filesystem and make sure PSDs change as expected
- [x] What happens if Photoshop has a file open, but we've deleted the board?
- [x] Test when `.link` points to a PSD that does not exist
- [x] What happens if we edit-in-photoshop a board _on a different scene_? _ANSWER: Currently, this will fail silently, unless save happens when scene is open_
- [ ] Should we stop watching PSDs that aren't part of the new selection? _ANSWER: No, for now we'll keep watching until 1) their layer is drawn on/unlinked or 2) the session ends_

If a user has a linked PSD open in Photoshop, closes Storyboarder and re-opens, and then re-saves the (still open) PSD, it won't update in Storyboarder. We could begin watching all links _when Storyboarder opens a project_, but we're not doing that yet. For now, you'll have to notice that it's not updating and choose "Open in Photoshop" again.

Fixes #654

---
FUTURE
- We only break the link on `pointerdown`. Would be better to break the link on _any_ activity that might change the canvas (e.g.: flip, paste)
